x-pack/solutions/security/test/security_solution_api_integration/test_suites/detections_response/rules_management/rule_management/basic_license_essentials_tier/coverage_overview.ts

```diff
@@ -249,6 +249,167 @@ export default ({ getService }: FtrProviderContext): void => {
+              coverage: {},
+              unmapped_rule_ids: [expectedRule1.id],
+              rules_data: {
+                [expectedRule1.id]: {
+                  activity: 'disabled',
+                  name: 'Linux SMP aarch64 GNU/Linux',
+                },
+              },
+            });
+          });
+
+          it('returns exact term matches on rule name (when using multiple terms)', async () => {
+            const expectedRule1 = await createRule(
+              supertest,
+              log,
+              getCustomQueryRuleParams({
+                rule_id: 'rule-1',
+                name: 'User-Agent rule1',
+              })
+            );
+
+            const expectedRule2 = await createRule(
+              supertest,
+              log,
+              getCustomQueryRuleParams({
+                rule_id: 'rule-2',
+                name: 'User Agent rule2',
+              })
+            );
+
+            const body1 = await getCoverageOverview(supertest, {
+              search_term: 'User-Agent rule1',
+            });
+
+            expect(body1).to.eql({
+              coverage: {},
+              unmapped_rule_ids: [expectedRule1.id],
+              rules_data: {
+                [expectedRule1.id]: {
+                  activity: 'disabled',
+                  name: 'User-Agent rule1',
+                },
+              },
+            });
+
+            const body2 = await getCoverageOverview(supertest, {
+              search_term: 'User Agent rule2',
+            });
+
+            expect(body2).to.eql({
+              coverage: {},
+              unmapped_rule_ids: [expectedRule2.id],
+              rules_data: {
+                [expectedRule2.id]: {
+                  activity: 'disabled',
+                  name: 'User Agent rule2',
+                },
+              },
+            });
+
+            const body3 = await getCoverageOverview(supertest, {
+              search_term: 'User Agent rule',
+            });
+
+            expect(body3).to.eql({
+              coverage: {},
+              unmapped_rule_ids: [],
+              rules_data: {},
+            });
+          });
+
+          it('Can find special character matches on rule name', async () => {
+            const expectedRule = await createRule(
+              supertest,
+              log,
+              getCustomQueryRuleParams({
+                rule_id: 'rule-1',
+                name: 'Rule name with a semi:colon',
+              })
+            );
+
+            const body = await getCoverageOverview(supertest, {
+              search_term: 'semi:colon',
+            });
+
+            expect(body).to.eql({
+              coverage: {},
+              unmapped_rule_ids: [expectedRule.id],
+              rules_data: {
+                [expectedRule.id]: {
+                  activity: 'disabled',
+                  name: 'Rule name with a semi:colon',
+                },
+              },
+            });
+          });
+
```

Comment:
Note that this has been through the Flaky Test runner x100 executions:

https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9545